### PR TITLE
Use cached hash function results in PHP 7.x  + disable all internals of tracing when its not needed 

### PR DESCRIPTION
**Pull request by pawelchcki**
### Description

This PR changes the implementation slightly to honor PHP 7.x cached hash calculations. 
In addition a new function has been implemented `dd_trace_disable_in_request` which is used to disable tracing when its no longer needed in order to  avoid spending CPU time unnecesarily

### Readiness checklist
- [x] [Changelog entry](docs/changelog.md) added, if necessary
- [x] Tests added for this feature/bug

### CODE REVIEW

File: bridge/dd_wrap_autoloader.php [hunk 1]
```diff
@@ -5,10 +5,12 @@
 require_once __DIR__ . '/functions.php';
 
 if (php_sapi_name() == 'cli' && getenv('APP_ENV') != 'dd_testing') {
+    dd_trace_disable_in_request();
```
Comment (pawelchcki) [inline bridge/dd_wrap_autoloader.php]: @labbati FYI

File: src/ext/dispatch.h [hunk 2]
```diff
@@ -12,6 +12,15 @@ typedef struct _ddtrace_dispatch_t {
     uint32_t acquired;
 } ddtrace_dispatch_t;
 
+typedef struct _ddtrace_lookup_data_t {
+#if PHP_VERSION_ID < 70000
+    const char *function_name;
+    uint32_t function_name_length;
+#else
+    zend_string *function_name;
```
Comment (SammyK) [inline src/ext/dispatch.h]: ❤️

File: src/ext/dispatch.c [hunk 3]
```diff
@@ -457,19 +465,21 @@ int default_dispatch(zend_execute_data *execute_data TSRMLS_DC) {
 }
 
 int ddtrace_wrap_fcall(zend_execute_data *execute_data TSRMLS_DC) {
-    const char *function_name = NULL;
-    uint32_t function_name_length = 0;
     DD_PRINTF("OPCODE: %s", zend_get_opcode_name(EX(opline)->opcode));
+    if (DDTRACE_G(disable) || DDTRACE_G(disable_in_current_request)) {
+        return default_dispatch(execute_data);
```
Comment (SammyK) [inline src/ext/dispatch.c]: One little ZTS thing. :)

```suggestion
        return default_dispatch(execute_data TSRMLS_CC);
```

Comment (pawelchcki) [reply src/ext/dispatch.c]: RIght, I'll also add compilation test to make sure this is caught by CI.

**Comment by pawelchcki** [comment]
Thanks @SammyK for the review! I've added ZTS compilation test on PHP(5.6) and fixed reported errors :D

**Pull request merged**